MockKit/MockPumpManager.swift

```diff
@@ -560,8 +578,10 @@ public final class MockPumpManager: TestingPumpManager {
     public func setMaximumTempBasalRate(_ rate: Double) { }
 
     public func syncBasalRateSchedule(items scheduleItems: [RepeatingScheduleValue<Double>], completion: @escaping (Result<BasalRateSchedule, Error>) -> Void) {
+        state.basalRateSchedule = BasalRateSchedule(dailyItems: scheduleItems, timeZone: self.status.timeZone)
+
         DispatchQueue.global().asyncAfter(deadline: .now() + .milliseconds(500)) {
-            completion(.success(BasalRateSchedule(dailyItems: scheduleItems, timeZone: self.status.timeZone)!))
+            completion(.success(self.state.basalRateSchedule!))
```

Comment:
[nit] Since this completion is invoked asynchronously 1/2 second later, could this become `nil` within that time period? If this has a chance of happening, then I suggest not force unwrapping or using a local variable.

Reply:
I'll revert this to `BasalRateSchedule(dailyItems: scheduleItems, timeZone: self.status.timeZone)!` but still store the results for other use.